pkg/cmd/release-payload-controller/batchjob_controller.go

```diff
@@ -0,0 +1,279 @@
+	releasePayloadClient    releasepayloadclient.ReleaseV1alpha1Interface
+
+	batchJobNamespace string
+	batchJobLister    batchv1listers.JobLister
+
+	eventRecorder events.Recorder
+
+	cachesToSync []cache.InformerSynced
+
+	queue workqueue.RateLimitingInterface
+}
+
+func NewBatchJobStatusController(
+	releasePayloadNamespace string,
+	releasePayloadInformer releasepayloadinformer.ReleasePayloadInformer,
+	releasePayloadClient releasepayloadclient.ReleaseV1alpha1Interface,
+	batchJobNamespace string,
+	batchJobInformer batchv1informers.JobInformer,
+	eventRecorder events.Recorder,
+) (*BatchJobStatusController, error) {
+	c := &BatchJobStatusController{
+		releasePayloadNamespace: releasePayloadNamespace,
+		releasePayloadLister:    releasePayloadInformer.Lister(),
+		releasePayloadClient:    releasePayloadClient,
+		batchJobNamespace:       batchJobNamespace,
+		batchJobLister:          batchJobInformer.Lister(),
+		eventRecorder:           eventRecorder.WithComponentSuffix("batchjob-status-controller"),
+		queue:                   workqueue.NewNamedRateLimitingQueue(workqueue.DefaultControllerRateLimiter(), "BatchJobStatusController"),
+	}
+
+	c.cachesToSync = append(c.cachesToSync, releasePayloadInformer.Informer().HasSynced)
+	c.cachesToSync = append(c.cachesToSync, batchJobInformer.Informer().HasSynced)
+
+	batchJobFilter := func(obj interface{}) bool {
+		if batchJob, ok := obj.(*batchv1.Job); ok {
+			if _, ok := batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+				return true
+			}
+		}
+		if tombstone, ok := obj.(cache.DeletedFinalStateUnknown); ok {
+			if batchJob, ok := tombstone.Obj.(*batchv1.Job); ok {
+				if _, ok = batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]; ok {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	batchJobInformer.Informer().AddEventHandler(cache.FilteringResourceEventHandler{
+		FilterFunc: batchJobFilter,
+		Handler: cache.ResourceEventHandlerFuncs{
+			AddFunc:    c.lookupReleasePayload,
+			UpdateFunc: func(old, new interface{}) { c.lookupReleasePayload(new) },
+			DeleteFunc: c.lookupReleasePayload,
+		},
+	})
+
+	return c, nil
+}
+
+func (c *BatchJobStatusController) lookupReleasePayload(obj interface{}) {
+	batchJob, ok := obj.(*batchv1.Job)
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to cast batchJob: %v", obj))
+		return
+	}
+
+	klog.V(4).Infof("Processing batchJob: '%s/%s'", batchJob.Namespace, batchJob.Name)
+
+	tag, ok := batchJob.Annotations[releasecontroller.ReleaseAnnotationReleaseTag]
+	if !ok {
+		utilruntime.HandleError(fmt.Errorf("unable to process batchJob, missing '%s' annotation", releasecontroller.ReleaseAnnotationReleaseTag))
+		return
+	}
+
+	releasePayloadKey := fmt.Sprintf("%s/%s", c.releasePayloadNamespace, tag)
+	klog.V(4).Infof("Queueing ReleasePayload: %s", releasePayloadKey)
+	c.queue.Add(releasePayloadKey)
+}
+
+func (c *BatchJobStatusController) Run(ctx context.Context) {
```

Comment:
(ctx, numberOfWorker int)

Reply:
Added support for workers to the controllers that process batch/v1 jobs and prowjobs.